modules/metagetta/src/cljdoc_analyzer/metagetta/clojurescript.clj

```diff
@@ -81,29 +80,38 @@
          (remove unreferenced-protocol?)
          (map (partial read-var source-path file vars)))))
 
-(def common-foreign-libs
-  "Some ClojureScript projects assume that particular foreign libs
-  have been provided. When these foreign libs are unknown to the analyzer
-  env an error will be thrown so we stub out commonly required names."
-  [{:provides ["react"], :file "intentionally/missing.js"}])
-
-(defn- analyze-file [file]
-  (let [opts  (-> {:foreign-libs common-foreign-libs}
-                  (cljs.closure/add-implicit-options))
-        state (cljs.env/default-compiler-env opts)]
+(defn- fakenize-namespace!
+  "For each required namespace which is a string, generate a fake js module in
+  the compiler environment.
+  Return the updated compiler environment, 'state'.
+
+  Rational:
+  Required namespaces that are strings correspond to js library used by the package
+  currently parsed. The clojurescript package generally doesn't involved their js
+  libraries. The js dependencies are usually managed by npm/yarn package managed.
+  https://github.com/cljdoc/cljdoc-analyzer/issues/18"
+  [state js-dependencies]
+  ;; fake all string requires since otherwise npm indexing is required
+  (doseq [dependency js-dependencies
+          :when (not (contains? (:js-dependency-index @state) dependency))]
+    (swap! state assoc-in [:js-dependency-index dependency] (gensym "fake$module")))
+  state)
+
+(defn- analyze-file [js-dependencies file]
+  (let [state
+        (fakenize-namespace! (cljs.env/default-compiler-env) js-dependencies)]
```

Comment:
I think instead of writing a mutating `fakenize-namespace!` function here we could also compute the value we want for `:js-dependency-index` and then use `merge` to make sure we're not overwriting any already existing information: 

```clj
(let [state (cljs.env/default-compiler-env)
      faked-js-deps (zipmap js-dependencies (repeatedly #(gensym "fake$module")))]
  (swap! state update :js-dependency-index #(merge fake-js-deps %))
  (ana/no-warn ,,,,))
```

This separates the construction of the `js-dependency-index` value a bit more from the side effecting change to the compiler state making it easier to test and play around with in REPL sessions.

Reply:
To keep the docstring of the `fakenize-namespace!` function around we could move this bit into a function like `fake-js-deps`

```clj
(defn fake-js-deps 
  "Explanatory docstring"
  [js-dependencies]
  (zipmap js-dependencies (repeatedly #(gensym "fake$module"))))
```